hw09_struct_validator/validator.go

```diff
@@ -1,17 +1,254 @@
 package hw09structvalidator
 
+import (
+	"errors"
+	"fmt"
+	"log"
+	"reflect"
+	"regexp"
+	"strconv"
+	"strings"
+)
+
 type ValidationError struct {
 	Field string
 	Err   error
 }
 
 type ValidationErrors []ValidationError
 
+var (
+	ErrUnspecified        = errors.New("unspecified")
+	ErrNotStruct          = errors.New("not a struct")
+	ErrNil                = errors.New("can't validate a nil value")
+	ErrValidation         = errors.New("validation error")
+	ErrValidationDefition = errors.New("validator definition error")
+)
+
+var (
+	opLen    = regexp.MustCompile(`^len:(\d+)$`)
+	opRegexp = regexp.MustCompile(`^regexp:(.+)$`)
+	opIn     = regexp.MustCompile(`^in:(.+)$`)
+	opMin    = regexp.MustCompile(`^min:(\d+)$`)
+	opMax    = regexp.MustCompile(`^max:(\d+)$`)
+)
+
 func (v ValidationErrors) Error() string {
```

Comment:
```suggestion
func (v *ValidationErrors) Error() string {
```
Лучше не миксовать pointer & value receivers